### PR TITLE
feature/add_endpoints

### PR DESCRIPTION
**Pull request by rosealti**
…ndpoints for company and data13f.

### CODE REVIEW

File: python/edgar_filing_searcher/api/routes/company.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def get_company():
 
 @company_blueprint.route('/company/<company_id>/edgarfiling/')
 def get_filings(company_id):
-    """Route for results for search by company id"""
+    """Route for results for search by company id and date"""
```
Comment (datdo) [inline python/edgar_filing_searcher/api/routes/company.py:32]: filings by company id and date

File: python/edgar_filing_searcher/api/routes/company.py [hunk 2]
```diff
@@ -45,3 +45,18 @@ def get_filings(company_id):
         )
 
     return jsonify(list(filings))
+
+
+@company_blueprint.route('/company/<company_id>/')
+def get_filings_from_company(company_id):
+    """Route for results for search by company id"""
```
Comment (datdo) [inline python/edgar_filing_searcher/api/routes/company.py:52]: company by company id

File: python/edgar_filing_searcher/api/routes/company.py [hunk 3]
```diff
@@ -45,3 +45,18 @@ def get_filings(company_id):
         )
 
     return jsonify(list(filings))
+
+
+@company_blueprint.route('/company/<company_id>/')
+def get_filings_from_company(company_id):
+    """Route for results for search by company id"""
+    filings = EdgarFiling.query.filter(EdgarFiling.cik_no.like(f"{company_id}%"))
```
Comment (datdo) [inline python/edgar_filing_searcher/api/routes/company.py:53]: cik_no should equal company id

File: python/edgar_filing_searcher/api/routes/company.py [hunk 4]
```diff
@@ -45,3 +45,18 @@ def get_filings(company_id):
...
+
+
+@company_blueprint.route('/company/<company_id>/')
+def get_filings_from_company(company_id):
+    """Route for results for search by company id"""
+    filings = EdgarFiling.query.filter(EdgarFiling.cik_no.like(f"{company_id}%"))
+    return jsonify(list(filings))
+
+
+@company_blueprint.route('/company/<company_id>/<accession_id>')
```
Comment (datdo) [inline python/edgar_filing_searcher/api/routes/company.py:57]: /company/<company_id>/filing/<filing_id>

File: python/edgar_filing_searcher/api/routes/company.py [hunk 5]
```diff
@@ -45,3 +45,18 @@ def get_filings(company_id):
...
+def get_filings_from_company_(company_id, accession_id):
+    """Route for results for search by accession no"""
```
Comment (datdo) [inline python/edgar_filing_searcher/api/routes/company.py:59]: doc string again.

File: python/edgar_filing_searcher/api/routes/company.py [hunk 6]
```diff
@@ -45,3 +45,18 @@ def get_filings(company_id):
...
+    data13f = Data13f.query. \
+        filter(Data13f.cik_no == company_id, Data13f.accession_no.like(f"{accession_id}%"))
```
Comment (datdo) [inline python/edgar_filing_searcher/api/routes/company.py:61]: equals instead of like

**Review by datdo — CHANGES_REQUESTED** [review]

File: python/edgar_filing_searcher/api/routes/company.py (hunk 1 above)
Comment (rosealti) [reply python/edgar_filing_searcher/api/routes/company.py]: Changed

File: python/edgar_filing_searcher/api/routes/company.py (hunk 2 above)
Comment (rosealti) [reply python/edgar_filing_searcher/api/routes/company.py]: Changed

File: python/edgar_filing_searcher/api/routes/company.py (hunk 3 above)
Comment (rosealti) [reply python/edgar_filing_searcher/api/routes/company.py]: Changed

File: python/edgar_filing_searcher/api/routes/company.py (hunk 5 above)
Comment (rosealti) [reply python/edgar_filing_searcher/api/routes/company.py]: Changed

File: python/edgar_filing_searcher/api/routes/company.py (hunk 6 above)
Comment (rosealti) [reply python/edgar_filing_searcher/api/routes/company.py]: Changed

File: python/edgar_filing_searcher/api/routes/routes.py [hunk 7]
```diff
@@ -1,47 +1,55 @@
...
+def after_request(response):
+    """Enables cross origin resource sharing"""
+    header = response.headers
+    header['Access-Control-Allow-Origin'] = '*'
+    return response
+
+
+@company_blueprint.route('/company/search')
+def search_company():
+    """Route for search results by company name"""
```
Comment (datdo) [inline python/edgar_filing_searcher/api/routes/routes.py:21]: search for companies by company name

File: python/edgar_filing_searcher/api/routes/routes.py [hunk 8]
```diff
@@ -1,47 +1,55 @@
...
+    company_name = request.args.get('q')
+    companies = Company.query.filter(Company.company_name.ilike(f"%{company_name}%"))
+
+    if company_name is None:
+        abort(400, description="Bad Requests")
```
Comment (datdo) [inline python/edgar_filing_searcher/api/routes/routes.py:26]: Bad Request

File: python/edgar_filing_searcher/api/routes/routes.py [hunk 9]
```diff
@@ -1,47 +1,55 @@
...
+
+    return jsonify(list(companies))
+
+
+@company_blueprint.route('/company/<company_id>/edgarfiling/')
+def get_filings_with_date(company_id):
+    """Route for search results of filings by company id and date"""
```
Comment (datdo) [inline python/edgar_filing_searcher/api/routes/routes.py:33]: Route for filings for the specified company, optionally filtered by date

File: python/edgar_filing_searcher/api/routes/routes.py [hunk 10]
```diff
@@ -1,47 +1,55 @@
...
+        filings = filings.filter(
+            EdgarFiling.filing_date <= datetime.strptime(end_date, date_format)
+        )
+
+    return jsonify(list(filings))
+
+
+@company_blueprint.route('/edgarfiling/<accession_no>/data/')
+def get_filings_by_accession_no(accession_no):
+    """Route for search results of filing by filing id"""
```
Comment (datdo) [inline python/edgar_filing_searcher/api/routes/routes.py:53]: Route for data for specified edgar filing.

File: python/tests/api/routes/test_routes.py [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+    response = client.get('/company/search')
+    assert response.status_code == 400
+
+
+def test_search_company_HasNoArguments_json(client):
+    response = client.get('/company/search')
+    assert response.get_json() is None
+
+
+def test_get_filings_with_date_responseCode(client):  # add date param
```
Comment (datdo) [inline python/tests/api/routes/test_routes.py:75]: add test for date behavior

**Review by datdo — CHANGES_REQUESTED** [review]

**Review by datdo — APPROVED** [review]

**Review by TheNorthEestern — APPROVED** [review]

**Pull request merged**